File: apps/mybookkeeper/frontend/src/shared/types/rent/rent-payment-application.ts
/** One payment's contribution to one charge. */
export interface RentPaymentApplication {
  transaction_id: string;
  paid_on: string;
  /** How much of the payment landed on this charge. */
  amount: string;
  payer_name: string | null;
  /** The payment's full value, which may exceed `amount` when one payment
   *  spans two periods. */
  payment_total: string;
}

File: apps/mybookkeeper/backend/app/schemas/rent/rent_payment_application.py
"""One payment's contribution to one charge."""
from __future__ import annotations

import datetime as _dt
import uuid
from decimal import Decimal

from pydantic import BaseModel


class RentPaymentApplication(BaseModel):
    transaction_id: uuid.UUID
    paid_on: _dt.date
    amount: Decimal
    payer_name: str | None = None
    # The payment's full value, which may exceed ``amount`` when a single
    # payment spans two periods. Surfaced so the UI can say "$375 of $375"
    # versus "$125 of $375 (rest applied to September)".
    payment_total: Decimal
